### ISSUE TITLE
Language Server wrongly reports that an expression cannot be entered

### ISSUE THREAD

**Issue by notdanilo** [enso-org/enso#1091]
### General Summary

Wrongly reported `executionContext/expressionValuesComputed` makes it impossible for IDE to enter nodes.

### Steps to Reproduce

1. Run IDE.
2. Type the following code in `Main.enso`:
```
add a b = a + b


add2 a b =
    add here a b
    100

add3 a b =
    add2 here a b
    50

main = 
    IO.println "Hello, World!"
    add3 here 1 2
    add here 1 2
    200
```
3. Select `add3 here 1 2` and press `ctrl + enter` to enter the node.
4. It fails.

When checking the network log, you will notice that all the updates from `executionContext/expressionValuesComputed` returns with null method pointers.
![image](https://user-images.githubusercontent.com/1069140/90708765-f9009180-e270-11ea-870d-2db900c707c0.png)

### Expected Result
`executionContext/expressionValuesComputed` with method pointers on expressions that can be entered.

### Actual Result
`executionContext/expressionValuesComputed` has no entry with actual method pointers.

### Other Resources

A Google Chrome websocket session is recorded in [network.zip](https://github.com/enso-org/enso/files/5100279/network.zip). To open it, open the developer tools, go to network and import the HAL file by pressing the up arrow icon.

### Enso Version

```
Enso Compiler and Runtime
Version:    0.1.0
Built with: scala-2.13.3 for GraalVM 20.1.0
Built from: main* @ 6da3b9252f68b0b10dccb4eec37f83295f42f95b
Running on: OpenJDK 64-Bit Server VM, GraalVM Community, JDK 11.0.7+10-jvmci-20.1-b02
            Linux 5.4.0-42-generic (amd64)
```

**Comment by 4e6** [comment]
Language Server returns `methodPointer: null` because it can not find the appropriate method in the suggestion database.

The database is purely syntactic and stores the method as `(Unnamed.Main, here, add)` while the runtime is able to resolve `here` and requests `(Unnamed.Main, Main, add)`.

The workaround is to prefix the methods explicitly.

```ruby
Main.add a b = a + b
```

**Comment by kustosz** [comment]
Whaaaaa? What do you mean purely syntactic? Does it not use the compiler?

**Comment by 4e6** [comment]
I mean, it indeed uses the compiler output, but it doesn't resolve the `here` references. 

__Update__: We discussed that there is a method resolution pass implemented in #1067, that should be used to replace the `here` references with the actual self type.

**Comment by mwu-tow** [comment]
Wasn't this working before with the previous method pointer structure?
